apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/PartitionOptions.cs

```diff
@@ -0,0 +1,85 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License").
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at 
+//
+// https://www.apache.org/licenses/LICENSE-2.0 
+//
+// Unless required by applicable law or agreed to in writing, software 
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and 
+// limitations under the License.
+
+using System.Threading;
+
+namespace Google.Cloud.Spanner.Data;
+
+/// <summary>
+/// An immutable class representing partition options.
+/// </summary>
+/// <remarks>
+/// <para>The specified values of `PartitionSizeBytes` and `MaxPartitions`
+/// are utilized in the RPC call while generating the partition tokens.</para>
+/// <para>The specified flag for `DataBoostEnabled` will be stored in the request property
+/// of the <see cref="CommandPartition"/> object and is used when executing the
+/// partition object.For more information see
+/// [ReadRequest DataBoostEnabled](https://cloud.google.com/dotnet/docs/reference/Google.Cloud.Spanner.V1/latest/Google.Cloud.Spanner.V1.ReadRequest#Google_Cloud_Spanner_V1_ReadRequest_DataBoostEnabled) and
+/// [ExecuteSQLRequest DataBoostEnabled](https://cloud.google.com/dotnet/docs/reference/Google.Cloud.Spanner.V1/latest/Google.Cloud.Spanner.V1.ExecuteSqlRequest#Google_Cloud_Spanner_V1_ExecuteSqlRequest_DataBoostEnabled).</para>
+/// </remarks>
+public sealed class PartitionOptions
+{
+    /// <summary>
+    /// Returns a default instance of <see cref="PartitionOptions"/>.
+    /// </summary>
+    public static PartitionOptions Default { get; } = new PartitionOptions(null, null, false);
+
+    /// <summary>
+    /// Returns a new instance with the same options as this one, but with the specified new value for <see cref="DataBoostEnabled" />.
+    /// </summary>
+    /// <returns>A new instance of <see cref="PartitionOptions"/> class.</returns>
```

Comment:
No need for the " class" part. (And if you *do* want it, we need "A new instance of the PartitionOptions class" rather than just "A new instance of PartitionOptions class.".) Same for ther other methods.